packages/protocol/engine/tsc/testframework.go

```diff
@@ -0,0 +1,110 @@
+}
+
+func NewTestFramework(test *testing.T, opts ...options.Option[TestFramework]) (t *TestFramework) {
+	return options.Apply(&TestFramework{
+		test: test,
+		mockAcceptance: &MockAcceptanceGadget{
+			blockAcceptedEvent: event.NewLinkable[*acceptance.Block, acceptance.Events, *acceptance.Events](),
+			acceptedBlocks:     make(map[models.BlockID]bool),
+		},
+	}, opts, func(t *TestFramework) {
+		t.TestFramework = tangle.NewTestFramework(
+			test,
+			tangle.WithTangleOptions(t.optsTangle...),
+		)
+
+		if t.optsIsBlockAcceptedFunc == nil {
+			t.optsIsBlockAcceptedFunc = t.mockAcceptance.IsBlockAccepted
+		}
+		if t.optsBlockAcceptedEvent == nil {
+			t.optsBlockAcceptedEvent = t.mockAcceptance.blockAcceptedEvent
+		}
+		if t.optsClock == nil {
+			t.optsClock = clock.NewClock(time.Now().Add(-5 * time.Hour))
+		}
+
+		if t.OrphanageManager == nil {
+			t.OrphanageManager = New(t.optsIsBlockAcceptedFunc, t.optsBlockAcceptedEvent, t.TestFramework.Tangle, t.optsClock, t.optsTSCManager...)
+		}
+
+	})
+}
+
+func (t *TestFramework) AssertExplicitlyOrphaned(expectedState map[string]bool) {
+	for alias, expectedOrphanage := range expectedState {
+		t.AssertBlock(alias, func(block *booker.Block) {
+			assert.Equal(t.test, expectedOrphanage, block.IsExplicitlyOrphaned(), "block %s is incorrectly orphaned", block.ID())
+		})
+	}
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region Options //////////////////////////////////////////////////////////////////////////////////////////////////////
+
+func WithTSCManagerOptions(opts ...options.Option[TSCManager]) options.Option[TestFramework] {
+	return func(tf *TestFramework) {
+		tf.optsTSCManager = opts
+	}
+}
+
+func WithTangleOptions(opts ...options.Option[tangle.Tangle]) options.Option[TestFramework] {
+	return func(tf *TestFramework) {
+		tf.optsTangle = opts
+	}
+}
+
+func WithBlockAcceptedEvent(blockAcceptedEvent *event.Linkable[*acceptance.Block, acceptance.Events, *acceptance.Events]) options.Option[TestFramework] {
+	return func(tf *TestFramework) {
+		tf.optsBlockAcceptedEvent = blockAcceptedEvent
+	}
+}
+func WithIsBlockAcceptedFunc(isBlockAcceptedFunc func(id models.BlockID) bool) options.Option[TestFramework] {
+	return func(tf *TestFramework) {
+		tf.optsIsBlockAcceptedFunc = isBlockAcceptedFunc
+	}
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// MockAcceptanceGadget mocks ConfirmationOracle marking all blocks as confirmed.
+type MockAcceptanceGadget struct {
+	blockAcceptedEvent *event.Linkable[*acceptance.Block, acceptance.Events, *acceptance.Events]
+	acceptedBlocks     map[models.BlockID]bool
+}
+
+// IsBlockAccepted mocks its interface function returning that all blocks are confirmed.
+func (m *MockAcceptanceGadget) IsBlockAccepted(blockID models.BlockID) bool {
+	return m.acceptedBlocks[blockID]
+}
```

Comment:
Should this maybe live inside the `acceptance` package?

Reply:
Will move it there after merging this PR and scheduler PR as they both use this mock.